### PR TITLE
Update cartopy version and fix recipe

### PR DESCRIPTION
**Pull request by iarspider**
Cartopy 0.17.0 works fine with proj 6

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-cartopy/package.py [hunk 1]
```diff
@@ -21,7 +22,8 @@ class PyCartopy(PythonPackage):
     depends_on('py-pyshp@1.1.4:',   type=('build', 'run'))
     depends_on('py-six@1.3.0:',     type=('build', 'run'))
     depends_on('geos@3.3.3:')
-    depends_on('proj@4.9.0:5')
+    depends_on('proj@4.9.0:5', when='@0.16.0')
+    depends_on('proj@6:',      when='@0.17.0')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-cartopy/package.py]: Are you sure PROJ 4 and 5 don't work with 0.17.0? The `setup.py` still lists 4.9.0 as the minimum PROJ version.

File: var/spack/repos/builtin/packages/py-cartopy/package.py [hunk 2]
```diff
@@ -21,7 +22,8 @@ class PyCartopy(PythonPackage):
...
 
     # optional dependecies
     depends_on('py-matplotlib@1.5.1:', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-cartopy/package.py]: I would add variants for these optional dependencies. See the `requirements` directory in the tarball for variant names and which dependencies to add for each variant. Also, `py-filelock` is now needed for testing as well.

File: var/spack/repos/builtin/packages/py-cartopy/package.py (hunk 1 above)
Comment (iarspider) [reply var/spack/repos/builtin/packages/py-cartopy/package.py]: No. I just wanted to keep the recipe backwards compatible.

File: var/spack/repos/builtin/packages/py-cartopy/package.py (hunk 2 above)
Comment (iarspider) [reply var/spack/repos/builtin/packages/py-cartopy/package.py]: @adamjstewart done. I'm not sure if I did `when` option for `py-pillow`: it is required for both `plotting` and `ows`.

File: var/spack/repos/builtin/packages/py-cartopy/package.py [hunk 3]
```diff
@@ -12,26 +12,34 @@ class PyCartopy(PythonPackage):
...
     # optional dependecies
-    depends_on('py-matplotlib@1.5.1:', type=('build', 'run'))
-    depends_on('gdal@1.10.0:+python',  type=('build', 'run'))
-    depends_on('py-pillow@1.7.8:',     type=('build', 'run'))
-    depends_on('py-pyepsg@0.2.0:',     type=('build', 'run'))
-    depends_on('py-scipy@0.10:',       type=('build', 'run'))
-    depends_on('py-owslib@0.8.11:',    type=('build', 'run'))
+    depends_on('py-matplotlib@1.5.1:', type=('build', 'run'), when='+plotting')
+    depends_on('gdal@1.10.0:+python',  type=('build', 'run'), when='+plotting')
+    depends_on('py-pillow@1.7.8:',     type=('build', 'run'), when='+ows+plotting')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-cartopy/package.py]: This should be split into 2 lines, one for each variant.

Comment (iarspider) [reply var/spack/repos/builtin/packages/py-cartopy/package.py]: Stupid me! Done.

File: var/spack/repos/builtin/packages/py-cartopy/package.py [hunk 4]
```diff
@@ -12,26 +12,35 @@ class PyCartopy(PythonPackage):
     homepage = "http://scitools.org.uk/cartopy/"
     url      = "https://github.com/SciTools/cartopy/archive/v0.16.0.tar.gz"
 
+    version('0.17.0', sha256='137642e63952404ec0841fa0333ad14c58fbbf19cca2a5ac6a38498c4b4998fb')
     version('0.16.0', sha256='cadf62434492c965220b37f0548bc58180466ad6894a1db57dbc51cd43467e5c')
 
     depends_on('py-setuptools@0.7.2:', type='build')
     depends_on('py-cython@0.15.1:',    type='build')
+    depends_on('py-futures',        type=('build', 'run'), when='^python@:2')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-cartopy/package.py]: I don't see this listed anywhere

**Comment by adamjstewart** [comment]
Flake8:
```
var/spack/repos/builtin/packages/py-cartopy/package.py:32: [W293] blank line contains whitespace
```

File: var/spack/repos/builtin/packages/py-cartopy/package.py (hunk 4 above)
Comment (iarspider) [reply var/spack/repos/builtin/packages/py-cartopy/package.py]: Indeed, not required for now. Will probably be required for the next version: https://github.com/SciTools/cartopy/blob/master/requirements/default.txt#L6

**Pull request merged**